Rationalize icon font set

andy-armstrong opened the pull request:
## [FEDX-34](https://openedx.atlassian.net/browse/FEDX-34)

I've made some clean ups to the edX icon font set:
* I've cleaned up the names to be more consistent
* I've implemented fall-through logic for handling icon synonyms
* I've cleaned up the documentation to use the actual icon names

I've manually verified that the icons all display in the doc site, and I've also verified that the edx-platform pipeline can now collectstatic without error.

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @dsjen
- [ ] Accessibility review: @clrux

FYI: @AlasdairSwan @mattdrayer 

### Post-review
- [ ] Squash commits
- [ ] Publish packages/doc site

andy-armstrong commented on pattern-library/sass/patterns/_icons.scss:
```diff
@@ -17,117 +17,114 @@
 
 $edx-icons:
-linkedin-square
+angle-down
```
Reviewers: I made this list alphabetical to make it easier to find if an icon exists.

clrux commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -137,7 +137,7 @@ info:               Icons are used to clearly and visually represent some inform
         </div>
         <div class="icon-classname">
-            Enter fullscreen
+            arrows-alt (Enter Fullscreen)
```
This icon isn't showing up for me.

andy-armstrong replied:
Ah, I renamed it to just "arrows" in the font definition. I think I'm going to change it to "fullscreen" in both.

clrux commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -899,7 +921,7 @@ info:               Icons are used to clearly and visually represent some inform
         </div>
         <div class="icon-classname">
-            Visible, Show
+            eye
```
Re: accessibility, aren't these icons also used to "hide" and "show/reveal" things? Aren't they used as actions in addition to states? If so, maybe we can make notes that indicate to have accessible text that represents the purpose of the icon.

andy-armstrong replied:
That's a good point. "eye" isn't really a very good semantic name. How about I call this one "show" and the other one "hide"? I could leave "eye" as a synonym too.

clrux commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -910,18 +932,18 @@ info:               Icons are used to clearly and visually represent some inform
         </div>
         <div class="icon-classname">
-            Hidden, Hide
+            eye-slash
```
Re: accessibility, aren't these icons also used to "hide" and "show/reveal" things? Aren't they used as actions in addition to states? If so, maybe we can make notes that indicate to have accessible text that represents the purpose of the icon.

clrux commented:
Regarding the icons for forms... if we use the replace method in our UXPL then these icons theoretically could be hidden entirely, because users would still interact with the actual form control. If this icon is being used _not_ as a form control (like, in documentation or something) then it should use the fallback text. But if it's being used in our enhanced form controls, it should be hidden all the way, otherwise users would hear "radio checked" twice - once as they interact with the actual (hidden) form control, and again when they get to the icon.

andy-armstrong commented:
@clrux I agree about the best practice for when to use which technique for icons. I think this is covered in the wiki: https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Icons. Did you want me to add something to my PR to address this?

andy-armstrong commented:
@clrux Thanks for the review. I've pushed up a commit to address your feedback.
@dsjen This is ready for your review now.

If possible I'd like to merge this tomorrow so that I can wrap up the edx-platform work that is dependent on this.

andy-armstrong commented:
@AlasdairSwan I was looking at the impact of this change on your icon usage in Programs. It looks like the only icon I've renamed that will affect you is icon-trash-o => icon-trash. This will only be an issue if we switch Programs to use the installed Pattern Library rather than using its own.

dsjen commented on pattern-library/sass/patterns/_icons.scss:
```diff
@@ -497,3 +482,9 @@ cc;
+
+@each $synonym, $edx-icon in $edx-icon-synonyms {
+    .icon-fallback-img .icon-#{$synonym} {
+    background: url('#{$edx-icons-font-path}/fallback-img/#{$edx-icon}.svg') center center no-repeat;
```
It might be nice to pull this line and [the similar one above ~482](https://github.com/edx/ux-pattern-library/pull/284/files#diff-edfc1e00417629e4a2621e84f5676c68R482) into a mixin so that if one changes for some reason the other will follow suit.

AlasdairSwan commented:
@andy-armstrong great, so I can update the programs stuff later to use the UXPL that is installed in studio but for now we can keep it as is.

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -126,29 +126,29 @@ info:               Icons are used to clearly and visually represent some inform
         <div class="icon-display">
             <span class="icon-fallback icon-fallback-text">
-                <span class="icon icon-arrows-alt" aria-hidden="true"></span>
+                <span class="icon icon-fullscreen" aria-hidden="true"></span>
```
Nit: perhaps `full-screen` instead

andy-armstrong replied:
I went back-and-forth on this when I added it. It seems that fullscreen is the convention used for the mode:

http://english.stackexchange.com/questions/162421/fullscreen-or-full-screen

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -372,35 +372,35 @@ info:               Icons are used to clearly and visually represent some inform
+                <span class="icon icon-error" aria-hidden="true"></span>
                 <span class="text">Error</span>
             </span>
         </div>
```
I can't comment on [408](https://github.com/edx/ux-pattern-library/pull/284/files#diff-8e6a0a47e453c2074d1a3b7fbd3c0280R408), but it looks like that should be updated to the lowercase name now.

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -411,7 +411,7 @@ info:               Icons are used to clearly and visually represent some inform
+                <span class="icon icon-warning" aria-hidden="true"></span>
                 <span class="text">Warning</span>
             </span>
         </div>
```
Same comment on updating to lowercase for [419](https://github.com/edx/ux-pattern-library/pull/284/files#diff-8e6a0a47e453c2074d1a3b7fbd3c0280R419).

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -624,7 +635,7 @@ info:               Icons are used to clearly and visually represent some inform
+            home
         </div>
     </div>
     <div class="example example-icon">
```
It looks like "External link" at line [649](https://github.com/edx/ux-pattern-library/pull/284/files#diff-8e6a0a47e453c2074d1a3b7fbd3c0280R649) should be updated to the "external-link" instead.

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -536,7 +536,18 @@ info:               Icons are used to clearly and visually represent some inform
+    <div class="example example-icon">
+        <div class="icon-display">
+            <span class="icon-fallback icon-fallback-text">
+                <span class="icon icon-cog" aria-hidden="true"></span>
```
This is duplicated at line 721 as the "settings" icon.  Can they be combined?

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -701,18 +712,18 @@ info:               Icons are used to clearly and visually represent some inform
         </div>
         <div class="icon-classname">
-            Settings, Configure
+            settings, cog, gear
```
Please see comment about duplicated icon at line 545.

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -877,7 +899,7 @@ info:               Icons are used to clearly and visually represent some inform
         </div>
         <div class="icon-classname">
-            Quote
+            quote-left
```
Not that it's necessary, but it seems odd that there isn't an analogous `quote-right` in the icon set.

andy-armstrong replied:
I wondered about this too. I'm not adding any new icons as part of this change, but we should consider if this is necessary in a future PR.

dsjen commented on pattern-library/sass/patterns/_icons.scss:
```diff
@@ -17,117 +17,118 @@
+warning;
 
+// Useful synonyms
+$edx-icon-synonyms: (
```
This is great!

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -547,7 +558,7 @@ info:               Icons are used to clearly and visually represent some inform
         </div>
         <div class="icon-classname">
-            Sort
+            sort
```
"unsorted" is listed in synonyms.  Should it be listed here too or removed from the synonyms?

dsjen commented on _posts/patterns/2015-04-15-icons.md:
```diff
@@ -547,7 +558,7 @@ info:               Icons are used to clearly and visually represent some inform
+            sort
         </div>
     </div>
     <div class="example example-icon">
```
Unfortunately, I can't comment on the exact line.  It looks like "Undo" (line 572) should be lowercase and also include "rotate-left" as a synonym (or "rotate-left" should be removed from the synonym list).

dsjen commented:
@andy-armstrong -- I'm wrapped up the review.  I found a few cases where the text wasn't updated, synonyms were missing, and there's a duplicated icon listed.

clrux commented:
@andy-armstrong I don't think you need to add anything to your PR for icons in forms. I was just thinking out loud. Good to know it's in the docs, too.

andy-armstrong commented:
@dsjen Thanks for the thorough review that caught so many little mistakes! I've addressed them all and pushed up a squashed commit ready to merge. I've also removed the version number updates as we discussed, as we'll do these as part of the release process going forward. The only trick is we'll have to remember all the changes in a given release in order to update the CHANGELOG.md file correctly.

@clrux Can you give me thumbs if you think this is ready to go.

clrux commented:
:+1:

dsjen commented:
LGTM! :star: :fire: :star:
